### PR TITLE
Tl 74 add warning components to common library

### PR DESCRIPTION
**Pull request by RVerma90**
Produced `useErrorCodes` hook to consume in other microfrontends.

Returns `error-code` data from referenceData.
If request for `error-code` has failed, defaults to FE error dictionary.

### CODE REVIEW

File: lib/hooks/tests/use-error-codes.test.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { renderHook } from '@testing-library/react-hooks';
+import { useReferenceData } from '../../api/reference-data/v1';
```
Comment (cloudratha) [inline lib/hooks/tests/use-error-codes.test.ts:2]:
```tsx
import { useReferenceData } from '@mtfh/common/lib/api/reference-data/v1';
```

File: lib/hooks/tests/use-error-codes.test.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+import locale from '../../locale';
+import { useErrorCodes } from '../use-error-codes';
+
+jest.mock('../../api/reference-data/v1');
```
Comment (cloudratha) [inline lib/hooks/tests/use-error-codes.test.ts:6]: Do you need to mock, or could you use `msw`?
@hackney/mtfh-test-utils 👀

File: lib/hooks/use-error-codes.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  useEffect(() => {
+    const fromErrorsDefaultData = Object.entries(errors).map((e, i) => ({
+      [e[0]]: e[1],
+    }));
+
+    const fromErrorsReferenceData =
+      data.map((item) => ({
+        [item.code]: item.value,
+      })) || [];
```
Comment (cloudratha) [inline lib/hooks/use-error-codes.ts:31]: Beautiful mate ❤️

File: lib/hooks/use-error-codes.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react';
+import { useReferenceData } from '../api/reference-data/v1';
+import locale from '../locale';
+
+export const useErrorCodes = () => {
+  const [errors, setErrors] = useState<any>(locale.hooks.errorMessages);
```
Comment (cloudratha) [inline lib/hooks/use-error-codes.ts:6]: Can you change this to `errorMessages`, I was a little confused to what was an error form the api and what the messages were.

File: lib/hooks/use-error-codes.ts (hunk 3 above)
Comment (RVerma90) [reply lib/hooks/use-error-codes.ts:31]: Arigato Gozaimasu

File: lib/hooks/use-error-codes.ts (hunk 4 above)
Comment (RVerma90) [reply lib/hooks/use-error-codes.ts:6]: Good shout, yep can change

File: lib/hooks/tests/use-error-codes.test.ts (hunk 1 above)
Comment (RVerma90) [reply lib/hooks/tests/use-error-codes.test.ts:2]: Replaced

File: lib/hooks/tests/use-error-codes.test.ts (hunk 2 above)
Comment (RVerma90) [reply lib/hooks/tests/use-error-codes.test.ts]: Replaced all together with `msw`

**Review by cloudratha — APPROVED** [review]

**Pull request merged**